src/pyspiffe/bundle/x509_bundle/x509_bundle_set.py

```diff
@@ -13,26 +13,28 @@
 class X509BundleSet(object):
     """X509BundleSet is a set of X509Bundles objects, keyed by trust domain."""
 
-    def __init__(self, bundles_map: Mapping[TrustDomain, X509Bundle]) -> None:
+    def __init__(self, bundles_map: Dict[TrustDomain, X509Bundle]) -> None:
         """Creates a new initialized X509BundleSet with the given X509Bundle objects keyed by TrustDomain.
 
         Args:
             bundles_map: A map object of X509Bundle objects keyed by TrustDomain to initialize the X509BundleSet.
         """
 
-        # create and initialize a dict, so it can be updated (as Mapping doesn't define __setitem__)
-        bundles = {}
-        for td, bundle in bundles_map.items():
-            bundles[td] = bundle
-        self._bundles = bundles
+        self.lock = threading.Lock()
+
+        if bundles_map:
+            self._bundles = bundles_map.copy()
+        else:
+            self._bundles = dict()
```

Comment:
nit: I think we should rename the parameter to `bundle_dict`.

Reply:
or maybe it could be just `bundles`, because why repeat the type in the name when it's already explicit, right?